### ISSUE TITLE
Freezing?

### ISSUE THREAD

**Issue by DakHarding** [Dicebar/Raven#99]
I'm having a bizarre issue with the latest update (one downloaded today from the Curse app 9-19-24) that's causing my screen/game to freeze for 1-2 seconds during movement and/or when attempting to loot dead mobs. 

I really only used 4 addons - Bartender, Masque, Grid2, and Raven. When I disabled everything and left Raven on, the freezes continue. Disable Raven, freezes stop. So, I'm not sure what's going on, but wanted to report.

Thanks! (retail version)

**Comment by Dicebar** [comment]
Which class(es) are you experiencing this problem with? Do you use spell lists?

**Comment by DakHarding** [comment]
I play all classes. Happens across all of them. No spell lists. Just a lot of custom conditions and bar groups across all of them.

**Comment by Dicebar** [comment]
So the good news is that my profile sounds very similar to yours, and I'm not having any issues.

The bad news is that I'm unsure what is causing your Raven to act up.

The last update only contained two fixes:
1. A fix for Raven_Options, the code for which is not accessed during regular play.
2. A fix the code that reads out spell tabs, which threw errors in certain situations.

I honestly don't see how either fix could be affecting your game like this, but that's no guarantee either.

Can you try downgrading to r682 to see if the problem persists? I'd like to exclude the most recent fixes as a cause of this problem.

**Comment by DakHarding** [comment]
I went back a few updates. When I went from r679 to r680, boom, freezing again. So r679 and earlier appeared to be fine.

**Comment by Dicebar** [comment]
That greatly narrows it down, thanks!

It's probably related to the bug in GH-97, I think that's from the same piece of code.

**Comment by Dicebar** [comment]
I found a bug that might have been the cause.

Can you see if `r684` resolves your problems?

**Comment by DakHarding** [comment]
It's better, thank you. It'll sorta lock up when I first enter the game and go to attack something for 1-2 seconds, but then after that initial freeze it seems to stop.

**Comment by Dicebar** [comment]
That sounds like the cache filling up. I'm not sure why it creates chokes like that, and I'd like to investigate that at some point.

Like, I've got the same problem when I want to add conditions to a bar. Every time I click a checkbox it'll freeze up for a few seconds. I think the problem is part in huge profiles, as I've got one profile that covers every single class and spec, but I fear that solving it is going to require a big rewrite.

Glad to hear it's all working now though!
